### PR TITLE
[Improvement] GT-248 restart non-scheduled pod

### PR DESCRIPTION
**Pull request by nikita-vanyasin**

### CODE REVIEW

**Review by jwierzbo — APPROVED** [review]

File: pkg/deployment/reconcile/plan_builder_member_pod_scheduling_failure.go [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+			continue
+		}
+
+		renderedPod, err := context.RenderPodForMember(ctx, context.ACS(), spec, status, m.Member.ID, imageInfo)
+		if err != nil {
+			l.Err(err).Warn("could not render pod for already created member")
+			continue
+		}
+
+		if r.isSchedulingParametersChanged(renderedPod.Spec, m.Member, context) {
```
Comment (ajanikow) [inline pkg/deployment/reconcile/plan_builder_member_pod_scheduling_failure.go:70]: If currently terminating (ConditionTypeTerminating), skip

File: CHANGELOG.md [hunk 2]
```diff
@@ -12,6 +12,8 @@
 - (Maintenance) Bump K8S Version to 1.24.16
 - (Maintenance) Bump K8S Version to 1.25.12
 - (Maintenance) Bump Go to 1.20.7
+- (Improvement) Restart pod if it could not be scheduled and related scheduling parameters changed
```
Comment (ajanikow) [inline CHANGELOG.md:15]: Single CHANGELOG Line

**Review by ajanikow — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 2 above)
Comment (nikita-vanyasin) [reply CHANGELOG.md]: done

File: pkg/deployment/reconcile/plan_builder_member_pod_scheduling_failure.go (hunk 1 above)
Comment (nikita-vanyasin) [reply pkg/deployment/reconcile/plan_builder_member_pod_scheduling_failure.go:70]: done

**Review by ajanikow — APPROVED** [review]

**Pull request merged**